src/SourceTokenList.ts

```diff
@@ -271,18 +269,22 @@ export default class SourceTokenList {
 
   /**
    * Allow iterating over the tokens in this list using e.g. `for (… of …)`.
-   *
-   * $FlowIssue (see facebook/flow#252)
    */
   [Symbol.iterator]() {
     let index = this.startIndex;
     let { endIndex } = this;
-    return () => {
+    return (): { done: boolean, value: SourceToken | null | undefined } => {
       if (index === endIndex) {
         return { done: true, value: undefined };
       } else {
         let result = { done: false, value: this.tokenAtIndex(index) };
-        index = index.next();
+        let nextIndex = index.next();
+
+        if (!nextIndex) {
+          throw new Error(`unexpected null index before the end index`);
+        }
+
+        index = nextIndex;
```

Comment:
This is an example of TypeScript being stricter than Flow. Here, TypeScript sees that `tokenAtIndex` doesn't allow `null` but `index` could be set to `null` in `index = index.next();`.